pyaerocom/stats/data_filters.py

```diff
@@ -0,0 +1,34 @@
+import numpy as np
+
+
+class FilterNaN:
+    """
+    Excludes data for which either data or ref_data is NaN.
+    """
+
+    def __call__(
+        self, data: np.ndarray, ref_data: np.ndarray, weights: np.ndarray | None
+    ) -> tuple[np.ndarray, np.ndarray, np.ndarray | None]:
+        mask = ~np.isnan(ref_data) * ~np.isnan(data)
+
+        return mask
+
+
+class FilterByLimit:
+    """
+    Filters out data which is outside of a lower and / or upper limit.
+    """
+
+    def __init__(self, lowlim: float | None, highlim: float | None):
+        self.lowlim = lowlim
+        self.highlim = highlim
+
+    def __call__(
+        self, data: np.ndarray, ref_data: np.ndarray, weights: np.ndarray | None
+    ) -> tuple[np.ndarray, np.ndarray, np.ndarray | None]:
+        if self.lowlim is not None:
+            valid1 = np.logical_and(data > self.lowlim, ref_data > self.lowlim)
```

Comment:
`valid1` is not defined if `self.lowlim is None` (same problem with `valid2`)

Reply:
Should be fixed